src/ardupilot/ArduPilotProtocol.cpp

```diff
@@ -0,0 +1,117 @@
+#include "ArduPilotProtocol.h"
+
+#include "../Constants.h"
+#include "../Globals.h"
+#include "../log.h"
+#include "../utils/json.h"
+
+#include <mutex>
+
+namespace net {
+namespace ardupilot {
+
+using json = nlohmann::json;
+using val_t = json::value_t;
+using namespace robot::types;
+using namespace navtypes;
+using std::placeholders::_1;
+using namespace net::websocket;
+
+ArduPilotProtocol::ArduPilotProtocol() {
+	initArduPilotServer(Globals::websocketServer);
```

Comment:
Can we take the websocket server as a parameter? It doesn't need to be hardcoded here